src/actions/reactions/bookmark.js

```diff
@@ -0,0 +1,49 @@
+import toastr from 'toastr';
+import axios from '../../utils/axiosInstance';
+
+import {
+  BOOKMARK,
+  BOOKMARK_ERROR,
+  BOOKMARK_ON_ALL,
+  UNBOOKMARK,
+  UNBOOKMARK_ERROR,
+  UNBOOKMARK_ON_ALL,
+} from '../types';
+
+export const bookmarkArticle = (slug, articlePage, id) => async (dispatch) => {
+  try {
+    const { data } = await axios.post(`/articles/${slug}/bookmarks`);
+
+    if (articlePage === 'allArticles') {
+      dispatch({ type: BOOKMARK_ON_ALL, payload: { data, id } });
+      return toastr.success('Bookmarked article successfully');
+    }
+
+    dispatch({ type: BOOKMARK, payload: data });
+
+    return toastr.success('Bookmarked article successfully');
+  } catch ({ response }) {
+    const errorMessage = response.data;
+    toastr.error('Please login or signup');
+    return dispatch({ type: BOOKMARK_ERROR, payload: errorMessage });
+  }
+};
+
+export const unbookmarkArticle = (articleId, articlePage) => async (dispatch) => {
+  try {
+    const { data } = await axios.delete(`/user/bookmarks/${articleId}`);
+
+    if (articlePage === 'allArticles') {
+      dispatch({ type: UNBOOKMARK_ON_ALL, payload: { data, articleId } });
+      return toastr.success('Un-bookmarked article successfully');
```

Comment:
Here too, 'You have un-booked the article' would make more sense. I would like to know what you think, though.